Added play/pause and track skip to buttons 2-4.

JeffreyO opened the pull request:
I've 'fixed' the bug that crashes the program when an ad plays due to a NULL reference.

I also cleaned up your Visual Studio Project files and altered some things to make it more convenient and easier to build later on.

Enjoy!

JeffreyO committed:
Added play/pause and track skip to buttons 2-4.

I've fixed the bug that crashes the program when an ad plays due to a NULL reference.

I also cleaned up your Visual Studio Project files and altered some tings to make it more convenient and easier to build later on.

Enjoy!

LPardue commented:
Thanks!

I'll take a closer look and hope to merge soon.

On 18 Jan 2018 00:19, "Jeffrey Ozanne" <notifications@github.com> wrote:

> I've fixed the bug that crashes the program when an ad plays due to a NULL
> reference.
>
> I also cleaned up your Visual Studio Project files and altered some tings
> to make it more convenient and easier to build later on.
>
> Enjoy!
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/LPardue/SpotifyStatusApplet/pull/7
> Commit Summary
>
>    - Added play/pause and track skip to buttons 2-4.
>
> File Changes
>
>    - *M* Binaries/Newtonsoft.Json.dll
>    <https://github.com/LPardue/SpotifyStatusApplet/pull/7/files#diff-0>
>    (0)
>    - *M* LICENSE
>    <https://github.com/LPardue/SpotifyStatusApplet/pull/7/files#diff-1>
>    (404)
>    - *M* README.md
>    <https://github.com/LPardue/SpotifyStatusApplet/pull/7/files#diff-2>
>    (82)
>    - *A* ReleaseFiles/LICENSE
>    <https://github.com/LPardue/SpotifyStatusApplet/pull/7/files#diff-3>
>    (202)
>    - *A* ReleaseFiles/License LGPL.txt
>    <https://github.com/LPardue/SpotifyStatusApplet/pull/7/files#diff-4>
>    (143)
>    - *A* ReleaseFiles/License MIT.txt
>    <https://github.com/LPardue/SpotifyStatusApplet/pull/7/files#diff-5>
>    (11)
>    - *A* ReleaseFiles/README.md
>    <https://github.com/LPardue/SpotifyStatusApplet/pull/7/files#diff-6>
>    (42)
>    - *M* SpotifyLocalAPI/Properties/AssemblyInfo.cs
>    <https://github.com/LPardue/SpotifyStatusApplet/pull/7/files#diff-7>
>    (72)
>    - *M* SpotifyLocalAPI/Responses.cs
>    <https://github.com/LPardue/SpotifyStatusApplet/pull/7/files#diff-8>
>    (166)
>    - *M* SpotifyLocalAPI/SpotifyAPI.cs
>    <https://github.com/LPardue/SpotifyStatusApplet/pull/7/files#diff-9>
>    (574)
>    - *M* SpotifyLocalAPI/SpotifyLocalAPI.csproj
>    <https://github.com/LPardue/SpotifyStatusApplet/pull/7/files#diff-10>
>    (150)
>    - *M* SpotifyStatusApplet.sln
>    <https://github.com/LPardue/SpotifyStatusApplet/pull/7/files#diff-11>
>    (73)
>    - *M* SpotifyStatusApplet/AboutBox.Designer.cs
>    <https://github.com/LPardue/SpotifyStatusApplet/pull/7/files#diff-12>
>    (332)
>    - *M* SpotifyStatusApplet/AboutBox.cs
>    <https://github.com/LPardue/SpotifyStatusApplet/pull/7/files#diff-13>
>    (252)
>    - *M* SpotifyStatusApplet/AboutBox.resx
>    <https://github.com/LPardue/SpotifyStatusApplet/pull/7/files#diff-14>
>    (704)
>    - *M* SpotifyStatusApplet/AppTrayIcon.cs
>    <https://github.com/LPardue/SpotifyStatusApplet/pull/7/files#diff-15>
>    (132)
>    - *M* SpotifyStatusApplet/ContextMenu.cs
>    <https://github.com/LPardue/SpotifyStatusApplet/pull/7/files#diff-16>
>    (150)
>    - *M* SpotifyStatusApplet/LcdGraphics.cs
>    <https://github.com/LPardue/SpotifyStatusApplet/pull/7/files#diff-17>
>    (600)
>    - *M* SpotifyStatusApplet/MediaPlayerDetails.cs
>    <https://github.com/LPardue/SpotifyStatusApplet/pull/7/files#diff-18>
>    (70)
>    - *M* SpotifyStatusApplet/Program.cs
>    <https://github.com/LPardue/SpotifyStatusApplet/pull/7/files#diff-19>
>    (613)
>    - *M* SpotifyStatusApplet/Properties/AssemblyInfo.cs
>    <https://github.com/LPardue/SpotifyStatusApplet/pull/7/files#diff-20>
>    (72)
>    - *M* SpotifyStatusApplet/Properties/Resources.Designer.cs
>    <https://github.com/LPardue/SpotifyStatusApplet/pull/7/files#diff-21>
>    (126)
>    - *M* SpotifyStatusApplet/Properties/Resources.resx
>    <https://github.com/LPardue/SpotifyStatusApplet/pull/7/files#diff-22>
>    (238)
>    - *M* SpotifyStatusApplet/SpotifyStatusApplet.csproj
>    <https://github.com/LPardue/SpotifyStatusApplet/pull/7/files#diff-23>
>    (352)
>    - *A* SpotifyStatusApplet/Thumbs.db
>    <https://github.com/LPardue/SpotifyStatusApplet/pull/7/files#diff-24>
>    (0)
>
> Patch Links:
>
>    - https://github.com/LPardue/SpotifyStatusApplet/pull/7.patch
>    - https://github.com/LPardue/SpotifyStatusApplet/pull/7.diff
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/LPardue/SpotifyStatusApplet/pull/7>, or mute the
> thread
> <https://github.com/notifications/unsubscribe-auth/AGRFtW9Jpfm8UVifsxLVUALvVvQQ5gf7ks5tLo4FgaJpZM4RiMXL>
> .
>

JeffreyO commented:
Fixed the bug is a bit of a stretch. I've added a try-catch block temporarily so that the program doesn't crash (And so I get debug output when the exception is thrown.). It still throws the "System.NullReferenceException" exception during the ad, though it still runs rather than crashing. When the program picks up correct track data after the ad, it resumes normal operation as if nothing had happened.

I'm looking for a way to completely eliminate the exception from happening by checking to see if any of the needed values are NULL and substituting in a place-holder value. It should be commited once I have the issue taken care of.

JeffreyO commented on SpotifyStatusApplet/SpotifyStatusApplet.csproj:
```diff
@@ -1,167 +1,187 @@
+    </PreBuildEvent>
+  </PropertyGroup>
+  <PropertyGroup>
+    <PostBuildEvent>xcopy /d /y "$(SolutionDir)\Binaries\*.dll" $(TargertDir)
```
TODO: Fix typo. $(TargertDir) needs to be $(TargetDir). Whoops.

JeffreyO commented on README.md:
```diff
@@ -1,40 +1,42 @@
-- [Spotify Local API](https://code.google.com/p/spotify-local-api/) - A very simple and small library that allows .NET developers to get track information, (un)pause spotify, play tracks, get cover art and more! 
-- [GammaJul LgLcd](http://gjlglcd.codeplex.com/) - A .NET wrapper around the Logitech SDK for G15/G19 keyboard screens. Supports raw byte sending, GDI+ drawing and rendering WPF elements onto the screen.
-- [Json.net](http://json.codeplex.com/) - Json.NET is a popular high-performance JSON framework for .NET.
+SpotifyStatusApplet
```
I believe the diff operation is failing due to changed file permissions. The GitHub application (for Windows 7) had issues synchronizing changes as well, so I had to upload the files to GitHub through the web interface.
